website/source/docs/guides/production-acls.html.md

```diff
@@ -0,0 +1,484 @@
+configuration parameters stored in the KV. Similarly, a "write" policy with the
+same prefix would allow you to update any keys that begin with "redis/".
+
+### Write Privileges for One Key
+
+``` 
+key "dashboard-app" {
+  policy = "write" 
+} 
+```
+
+In the above example, we are allowing read and write privileges to the
+dashboard-app key. This allows for `get`, `delete`,  and `put` operations. 
+
+This type of token would allow an application to update and read a value in the
+KV store. It would also be useful for operators who need access to set specific
+keys. 
+
+### Read Privileges for One Key
+
+``` 
+key "counting-app" { 
+  policy = "read" 
+} 
+```
+
+In the above example, we are setting a read privileges for a single key,
+“counting-app”. This allows for only `get` operations. 
+
+This type of token allows an application to simply read from a key to get the
+value. This is useful for configuration parameter updates.
+
+If you want to grant a service read access to a key in the data store, the key
+rule should be added to the service's policy. For example, if the dashboard app
+also needs read privileges to the "dashboard-app" key, than we would update the
+existing policy with the new rule.
+
+```
+# dashboard-policy.hcl
+service "dashboard" { 
+  policy = "write" 
+  } 
+key "dashboard-app" { 
+  policy = "read"
+  } 
+```
+
+You do not need to create a new token or update the service definition again,
+since saving the policy will automatically update the token with the new
+privileges.
+
+## Consul UI Token
+
+Once you have bootstrapped the ACL system, access to the UI will be limited.
+The anonymous token grants UI access if no [default
+token](https://www.consul.io/docs/agent/options.html#acl_tokens_default) is set
+on the agents, and all operations will be denied, including viewing nodes and
+services. 
+
+You can re-enable UI features (with flexible levels of access) by creating and
+distributing tokens to operators. Once you have a token, you can use it in the
+UI by adding it to the "ACL" page:
+
+![Access Controls](/assets/images/guides/access-controls.png "Access Controls")
+
+After saving a new token, you will be able to see your tokens.
+
+![Tokens](/assets/images/guides/tokens.png "Tokens")
+
+The browser stores tokens that you add to the UI. This allows you to distribute
+different levels of privileges to operators. Like other tokens, it's up to the
+operator to decide the per-token privileges. 
+
+Below is an example of policy that
+will allow the operator to have read access to the UI for services, nodes,
+key/values, and intentions. You need to have  "acl = read"  to view policies
+and tokens. Otherwise you will not be able to access the ACL section of the UI,
+not even to view the token you used to access the UI.
+
+```
+# operator-ui.hcl
+service_prefix "" { 
+  policy = "read" 
+  } 
+key_prefix "" { 
+  policy = "read" 
+  }
+node_prefix "" { 
+  policy = "read" 
+  }
+intention_prefix "" { 
```

Comment:
There is no intention* rule resource. For now this is a subset of a service resource.

```
service_prefix "" {
   policy = "read"
}
```

This implicitly grants **read** on intentions for all services matching the "" prefix.

```
service "foo" {
   policy = "write"
```

Implicitly grants **read** on intentions for the foo service.

```
service "foo" {
   # could also be "deny" or "write"
   policy = "read"
   intentions = "write"
}
```

This has an explicit intentions policy set which happens to exceed the main policy. This will allow writing intentions to allow traffic **to** service foo.

Reply:
Good catch! For simplicity I'm going to remove the resource and let the users read more to figure out per service intentions policies.